snapcraft/internal/repo.py

```diff
@@ -115,28 +119,130 @@ def __init__(self, package_name):
         self.package_name = package_name
 
 
+class _AptCache:
+
+    def __init__(self, cache_dir, deb_arch, *,
+                 sources_list=None, use_geoip=False):
+        self._cache_dir = cache_dir
+        self._deb_arch = deb_arch
+        self._sources_list = sources_list
+        self._use_geoip = use_geoip
+
+    def _setup_apt(self, download_dir):
+        apt.apt_pkg.config.set("Dir::Cache::Archives", download_dir)
+
+        # Do not install recommends
+        apt.apt_pkg.config.set('Apt::Install-Recommends', 'False')
+
+        # Make sure we always use the system GPG configuration, even with
+        # apt.Cache(rootdir).
+        for key in 'Dir::Etc::Trusted', 'Dir::Etc::TrustedParts':
+            apt.apt_pkg.config.set(key, apt.apt_pkg.config.find_file(key))
+
+        apt.apt_pkg.config.clear('APT::Update::Post-Invoke-Success')
+
+        self.progress = apt.progress.text.AcquireProgress()
+        if not os.isatty(1):
+            # Make output more suitable for logging.
+            self.progress.pulse = lambda owner: True
+            self.progress._width = 0
+
+    def _setup_sources_list(self, rootdir):
+        if self._use_geoip or self._sources_list:
+            release = platform.linux_distribution()[2]
+            sources_list = _format_sources_list(
+                self._sources_list, self._deb_arch, self._use_geopip, release)
+        else:
+            sources_list = _get_local_sources_list()
+
+        sources_list_digest = hashlib.sha384(
+            sources_list.encode(sys.getfilesystemencoding())).hexdigest()
+
+        cache_dir = os.path.join(self._cache_dir, sources_list_digest)
+        self.apt_cache_dir = os.path.join(cache_dir, 'apt')
```

Comment:
This is used only in this method. Any reason for making it a class attribute instead of a local var?